src/components/App/App.js

```diff
@@ -62,12 +64,21 @@ class App extends Component {
       userInfo: {
         name: '',
         email: '',
-        purpose: ''
+        purpose: '',
+        favorites: []
       },
       isLoggedIn: false
     })
   }
 
+  updateFavorites = (newFavorite) => {
+    this.setState({
+      userInfo: {
+        favorites: [...this.state.userInfo.favorites, newFavorite]
+      }
+    })
+  }
+
```

Comment:
Called by ListingContainer, once the ListingContainer has found the correct listing to add to favorites.